### ISSUE TITLE
containerd crashes when memory usage is high

### ISSUE THREAD

**Issue by sabbatical66** [cruizba/ubuntu-dind#34]
Thanks for this repo, I haven't been able to figure it out myself for a long time! During usage, I encountered an issue: when I start many new containers and the total memory reaches its limit, containerd crashes and automatically restarts. This causes dockerd to lose connection, and all containers started by the previous containerd process are terminated. 
Normally, containerd or dockerd should have higher priority and should kill individual running containers before an out-of-memory (OOM) situation occurs. I don't know how to handle this issue myself.

**Comment by cruizba** [comment]
Mmm are you sure this isn't a problem of containerd itself? This repository just install docker in a standard ubuntu image. I have no control over it.

**Comment by sabbatical66** [comment]
> Mmm are you sure this isn't a problem of containerd itself? This repository just install docker in a standard ubuntu image. I have no control over it.

Yes, I ve seen the Dockerfile. However in this case, what happens inside the container is indeed different from what happens on the physical machine...
